Source/WebCore/bindings/js/JSDOMPromiseDeferred.h

```diff
@@ -164,6 +164,12 @@ class DeferredPromise : public DOMGuarded<JSC::JSPromise> {
     WEBCORE_EXPORT void reject(Exception, RejectAsHandled = RejectAsHandled::No);
     WEBCORE_EXPORT void reject(ExceptionCode, const String& = { }, RejectAsHandled = RejectAsHandled::No);
 
+    bool wasRejected() const
+    {
+        auto* lexicalGlobalObject = globalObject();
+        return deferred()->status(lexicalGlobalObject->vm()) == JSC::JSPromise::Status::Rejected;
```

Comment:
Do we need to ASSERT deferred() and globalObject() like in some other places?

Reply:
I can add them.